src/main/groovy/io/gatling/gradle/GatlingEnterpriseStartTask.groovy

```diff
@@ -9,45 +11,72 @@ import org.gradle.api.tasks.TaskExecutionException
 
 @CacheableTask
 class GatlingEnterpriseStartTask extends DefaultTask {
+    private static final String BATCH_MODE = "batchMode"
 
     @Internal
     Closure simulations
 
     @TaskAction
     void publish() {
         def gatling = project.extensions.getByType(GatlingPluginExtension)
-        EnterpriseClient enterpriseClient = gatling.enterprise.getEnterpriseClient(project.version.toString())
 
         def systemProps = gatling.enterprise.systemProps ?: [:]
 
-        if (!gatling.enterprise.simulationId && !gatling.enterprise.simulationClass) {
-            throw new TaskExecutionException(this,
-                new IllegalArgumentException("You need to either configure gatling.enterprise.simulationId if you want to start an existing simulation," +
-                "or gatling.enterprise.simulationClass if you want to create a new simulation")
-            )
-        }
-
         def simulationAndRunSummary
-        if (gatling.enterprise.simulationId) {
-            simulationAndRunSummary = enterpriseClient.startSimulation(gatling.enterprise.simulationId, systemProps, inputs.files.singleFile)
-        } else {
-            simulationAndRunSummary = enterpriseClient.createAndStartSimulation(
-                gatling.enterprise.teamId,
-                project.group.toString(),
-                project.name,
-                gatling.enterprise.simulationClass,
-                systemProps, inputs.files.singleFile)
-            def simulation = simulationAndRunSummary.simulation
-            getLogger().info("""
+
+        def batchMode = project.hasProperty(BATCH_MODE) ? Boolean.parseBoolean(project.getProperties().get(BATCH_MODE).toString()) : gatling.enterprise.batchMode
+
+        if (batchMode || gatling.enterprise.simulationId) {
+            if (!gatling.enterprise.simulationId && !gatling.enterprise.simulationClass) {
+                throw new TaskExecutionException(this,
+                    new IllegalArgumentException("You need to either configure gatling.enterprise.simulationId if you want to start an existing simulation," +
+                        "or gatling.enterprise.simulationClass if you want to create a new simulation")
+                )
+            }
+            EnterprisePlugin enterprisePlugin = gatling.enterprise.initEnterprisePlugin(project.version.toString(), getLogger())
+
+            if (gatling.enterprise.simulationId) {
+                simulationAndRunSummary = enterprisePlugin.uploadPackageAndStartSimulation(gatling.enterprise.simulationId, systemProps, inputs.files.singleFile)
+            } else {
+                getLogger().info("No simulationId configured, creating a new simulation")
```

Comment:
nit: you could define logger at the beginning instead of calling the getter everytime

Reply:
I find it necessary, as it is the goal of a getter

Reply:
why defining it?
As you are in groovy, you can directly use it

```suggestion
                logger.info("No simulationId configured, creating a new simulation")
```